### PR TITLE
Add a monkey patch to render Super Scaffolding model collections

### PR DESCRIPTION
**Pull request by gazayas**
Joint PRs
- https://github.com/bullet-train-co/bullet_train-super_scaffolding/pull/83
- https://github.com/bullet-train-co/bullet_train/pull/467

I explain what this monkey patch is for in https://github.com/bullet-train-co/bullet_train-base/issues/129.

### CODE REVIEW

**Comment by andrewculver** [comment]
@gazayas This is awesome. Can you resubmit this as a PR to `bullet_train-core`?

**Comment by andrewculver** [comment]
Added manually.

File: app/helpers/theme_helper.rb [hunk 1]
```diff
@@ -34,6 +34,25 @@ def find(name, prefixes = [], partial = false, keys = [], options = {})
...
+      def _to_partial_path # :nodoc:
+        @_to_partial_path ||= begin
+          element = ActiveSupport::Inflector.underscore(ActiveSupport::Inflector.demodulize(name))
+          collection = ActiveSupport::Inflector.tableize(name)
+          "#{collection.split("/").last}/#{element}"
+        end
+      end
+    end
+  end
+end
```
Comment (kaspth) [inline app/helpers/theme_helper.rb:54]: It seems risky to me to bury a monkey patch in here that redefines partial path generation across a whole Rails app.

Rails has a public API to override which is `to_partial_path` as an instance method, so I'm confused why we can't get away with the code below here?

```ruby
class Scaffolding::AbsolutelyAbstract::CreativeConcept
  def to_partial_path
    "creative_concepts/creative_concept"
  end
end
```

Are there other scaffolding models that need a similar method override?

Comment (andrewculver) [reply app/helpers/theme_helper.rb:54]: @kaspth You’re 100% right. I ended up running into a bunch of issues with this and have already removed it.

Comment (kaspth) [reply app/helpers/theme_helper.rb:54]: @andrewculver ah ok, got it.

@gazayas if you're interested, let me know if you want some pointers on how I think about monkey patches and integrating with Rails — hit me up in the BulletTrain internal Discord chat?

Comment (gazayas) [reply app/helpers/theme_helper.rb:54]: Sorry guys! @kaspth, I'll definitely send you a message soon!

**Pull request closed without merging**